src/renderers/dom/fiber/ReactDOMFiber.js

```diff
@@ -107,14 +119,19 @@ var DOMRenderer = ReactFiberReconciler({
     domElement : Instance,
     oldProps : Props,
     newProps : Props,
-    internalInstanceHandle : Object
+    rootContainerInstance : Container,
+    internalInstanceHandle : Object,
   ) : void {
-    var type = domElement.tagName.toLowerCase(); // HACK
-    var root = document.documentElement; // HACK
+    // TODO: we normalize here because DOM renderer expects tag to be lowercase.
+    // We can change DOM renderer to compare special case against upper case,
+    // and use tagName (which is upper case for HTML DOM elements). Or we could
+    // let the renderer "normalize" the fiber type so we don't have to read
+    // the type from DOM. However we need to remember SVG is case-sensitive.
+    var tag = domElement.tagName.toLowerCase();
```

Comment:
This is likely going to be an issue to do for every update. We might want to consider just banning non-lowercase uses instead.